### PR TITLE
Add issue and PR templates

### PR DESCRIPTION
**Pull request by nikhita**
- Since we use Prow, it will be easier to handle release notes and labels using the templates.
- Since this is an open source project, it's better to have templates.

/assign @alvaroaleman

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/enhancement.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+name: Enhancement Request
+about: Suggest an enhancement to tail
+
+---
+<!-- Please only use this template for submitting enhancement requests -->
+
+
+**What would you like to be added**:
+
+**Why is this needed**:
```
Comment (nikhita) [inline .github/ISSUE_TEMPLATE/enhancement.md:11]: I used the one in kubernetes/kubernetes https://github.com/kubernetes/kubernetes/blob/master/.github/ISSUE_TEMPLATE/enhancement.md. Should this be "User story" instead?

Comment (alvaroaleman) [reply .github/ISSUE_TEMPLATE/enhancement.md:11]: I think this is fine as-is, one can just put his or her user story below it

**Pull request merged**